pkg/controller/pxc/status.go

```diff
@@ -103,7 +104,26 @@ func (r *ReconcilePerconaXtraDBCluster) updateStatus(cr *api.PerconaXtraDBCluste
 		}
 
 		cr.Status.ProxySQL = proxyStatus
-		cr.Status.Host = cr.Name + "-" + "proxysql"
+
+		cr.Status.Host = cr.Name + "-" + "proxysql." + cr.Namespace
+		if cr.Spec.ProxySQL.ServiceType != nil && *cr.Spec.ProxySQL.ServiceType == corev1.ServiceTypeLoadBalancer {
```

Comment:
Not mandatory, but consider changing cr.Spec.ProxySQL.ServiceType to the non-pointer type. There is no reason for it to be a pointer. We always dereference it thru all the code.

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/373/commits/d6e19c57762498e95ec902e214a382db7af0e979